packages/crayons-core/src/components/popover/popover.tsx

```diff
@@ -110,64 +119,30 @@ export class Popover {
   @Method()
   async show() {
     if (!this.isOpen) {
-      this.sameWidth &&
-        (this.popperDiv.style.width =
-          String(this.triggerRef.getBoundingClientRect().width) + 'px');
-
-      // Create popper instance if it's not available
-      !this.popperInstance && this.createPopperInstance();
-
-      this.popperDiv.setAttribute('data-show', '');
-      // Enable the event listeners
-      this.popperInstance.setOptions((options) => ({
-        ...options,
-        modifiers: [
-          ...options.modifiers,
-          { name: 'eventListeners', enabled: true },
-        ],
-      }));
-      this.popperInstance.update();
-      if (this.trigger !== 'hover') {
-        this.overlay.style.display = 'block';
-      }
-      this.isOpen = !this.isOpen;
-      if (this.contentRef?.tagName === 'FW-LIST-OPTIONS') {
-        const listOptionsElement = this.contentRef as HTMLFwListOptionsElement;
-        listOptionsElement.scrollToLastSelected();
+      clearTimeout(this.timerId);
+      if (this.showAfter > 0) {
+        this.timerId = setTimeout(
```

Comment:
Can you create a function called delay 

```
async delay(ms) {  return new Promise((res) => { this.timerId = setTimeout(res,ms))  }}
```
and use this function in `show` method. 

```
if(!this.isOpen) {
clearTimeout(this.timerId);

if(this.showAfter> 0) await delay(this.showAfter)
```
.....
you can keep the rest of the code here instead of moving it to an another function

}

Reply:
Done